### PR TITLE
Added new samples 

### PR DESCRIPTION
**Pull request by ghermans**
### 1. Authentication
- Sample oAuth flow.

### 2. Groups
- Create a new group
- Retrieve all groups

### 3. Tickets

- Create a new ticket.
- Create a new ticket with the requester's email address, if the requester's identity doesn't exist.  
- Retrieve all tickets.
- Retrieve ticket details
- Retrieve tickets from the end-user email address   
- Update a ticket
- Delete a ticket
 
### 4. Users

- Create a new end-user
- Retrieve all end-users
- Search end-user

### CODE REVIEW

**Comment by joseconsador** [comment]
@ghermans Thanks! We can do PSR-2 cleanup on a separate pull request.

File: samples/tickets/getTickets.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+ $subdomain = "subdomain";
+ $username  = "email@example.com";
+ $token     = "6wiIBWbGkBMo1mRDMuVwkw1EPsNkeUj95PIz2akv";
+
+ $client = new ZendeskAPI($subdomain);
+ $client->setAuth('basic', ['username' => $username, 'token' => $token]);
+
+try {
+    $id   = "1";
```
Comment (samgavinio) [inline samples/tickets/getTickets.php]: Unused variable.

**Comment by ghermans** [comment]
@joseconsador @samgavinio ,  this should be ready for review :)

**Comment by ghermans** [comment]
@joseconsador @samgavinio @jmramos02 any idea when this will be merged? thanks in advance

**Pull request merged**